### PR TITLE
[Merged by Bors] - chore: Golf `Set.pair_eq_pair_iff`

### PR DESCRIPTION
**Pull request by YaelDillies**
Resolve a porting note

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Basic.lean
Comment (YaelDillies) [inline Mathlib/Data/Set/Basic.lean:1378]:
```suggestion
  aesop (add simp [subset_antisymm_iff, insert_subset_iff])
```
is twice as slow, cc @JLimperg

**Review by grunweg — APPROVED** [review]

**Comment by ADedecker** [comment]
Thanks! Delegating just in case you want to discuss the aesop issue before merging.

bors d+

**Comment by YaelDillies** [comment]
The aesop issue is non-blocking. Just wanted Jannis to have a chance to look!

maintainer merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**
